src/Microsoft.DocAsCode.Common/Git/GitUtility.cs

```diff
@@ -59,7 +59,31 @@ public static GitRepoInfo GetRepoInfo(string directory)
                 throw new GitException("Can't find git command in current environment");
             }
 
-            return Cache.GetOrAdd(directory, GetRepoInfoCore);
+            if (IsGitRoot(directory))
+            {
+                return Cache.GetOrAdd(directory, GetRepoInfoCore);
+            }
+
+            var parentDirInfo = Directory.GetParent(directory);
+            return Cache.GetOrAdd(directory, GetRepoInfo(parentDirInfo?.FullName));
```

Comment:
>GetRepoInfo [](start = 45, length = 11)

This recursion will cause more calling ExistGitCommand(). Extract a GetRepoInfoCore() or put this check somewhere else?